tests/integration/test_charm.py

```diff
@@ -212,13 +216,12 @@ async def test_setup_discourse(
     # Extract the CSRF token
     parsed_admin: BeautifulSoup = BeautifulSoup(response.content, features="html.parser")
     assert parsed_admin.head
-    csrf_token = parsed_admin.head.find("meta", attrs={"name": "csrf-token"}).get("content")  # type: ignore
+    csrf_token = parsed_admin.head.find("meta", attrs={"name": "csrf-token"}).get(  # type: ignore
+        "content"
+    )
```

Comment:
Generic question, will black complain if we return before `.get(` ?
I don't find this kind of indentation very elegant and readable... but perhaps that's the python way

Reply:
Good question - unfortunately black reformatted it this way, I set it differently, though without luck. If it wasn't for the type comment, it would fit in one line as well, perhaps that's confusing black in this case?

Reply:
Well this suggests to use `\`
https://stackoverflow.com/questions/8683178/chained-method-calls-indentation-style-in-python
=> Which I like for readability

But PEP8 specifies:
https://peps.python.org/pep-0008/#maximum-line-length
`The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.`

Which would suggest this:
```python
csrf_token = (parsed_admin.head.find("meta", attrs={"name": "csrf-token"}) # type: ignore
        .get( "content"))
```
Which I don't find very nice

@jdkandersson ?

Reply:
I read it differently, it implies what `black` is doing by breaking on the opening bracket of `.get(`, generally speaking on formatting - whatever black does is right 😉

Reply:
note the `inside` in `Python’s implied line continuation inside parentheses`

Reply:
Depending on which part the `# type: ignore` applies I think you can trick black into breaking somewhere else, although I usually do this through trial and error. E.g., you could try this, I'm not sure what black would do with it though:

```Python
csrf_token = parsed_admin.head.find(  # type: ignore
    "meta", attrs={"name": "csrf-token"}).get("content"
)
```

Reply:
Anyway, the point of using black is so we don't need to debate formatting like this and just use whatever it does

Reply:
> Depending on which part the `# type: ignore` applies I think you can trick black into breaking somewhere else, although I usually do this through trial and error. E.g., you could try this, I'm not sure what black would do with it though:
> 
> ```python
> csrf_token = parsed_admin.head.find(  # type: ignore
>     "meta", attrs={"name": "csrf-token"}
> ).get("content")
> ```

Using that example, black re-formats it to the above (what is already commited). At this point I'm fine with black doing its thing. I don't have more suggestions.

Reply:
Well I don't find that black is doing something very readable in that case, having a single string and parenthesis on their own line look ugly to me.
```python
csrf_token = parsed_admin.head.find(  # type: ignore
    "meta", attrs={"name": "csrf-token"}
).get("content")
```

This formatting is way better and more readable. And for sure we can trust black to avoid formatting discussion, but these tools aren't the alpha and the omega... they're just using algorithm and could produce something that can be improved while still being valid :)

Reply:
Based on the feedback I don't know what to do next. I don't have approvals either, so is there a suggestion I can try and implement?

Reply:
I'm in favor of the formatting I quoted in the previous comment

Reply:
Yes, but

```shell
merkata@mastodon:~/dev/discourse-k8s-operator$ grep "csrf_token =" tests/integration/test_charm.py -A3
    csrf_token = parsed_admin.head.find(  # type: ignore
        "meta", attrs={"name": "csrf-token"}
    ).get("content")

merkata@mastodon:~/dev/discourse-k8s-operator$ black tests/integration/test_charm.py 
reformatted tests/integration/test_charm.py

All done! ✨ 🍰 ✨
1 file reformatted.
merkata@mastodon:~/dev/discourse-k8s-operator$ grep "csrf_token =" tests/integration/test_charm.py -A3
    csrf_token = parsed_admin.head.find("meta", attrs={"name": "csrf-token"}).get(  # type: ignore
        "content"
    )
```